test/mox_test.exs

```diff
@@ -385,6 +403,26 @@ defmodule MoxTest do
       message = ~r"expected CalcMock.add/2 to be invoked 2 times but it was invoked once"
       assert_raise Mox.VerificationError, message, &verify!/0
     end
+
+    test "verifies mocks are over-called for the current process in global mode" do
+      start_supervised!({Task.Supervisor, name: MoxTests.TaskSupervisor})
+      set_mox_global()
+
+      verify!()
+      expect(CalcMock, :add, 1, fn x, y -> x + y end)
+
+      message = ~r"expected CalcMock.add/2 to be invoked once but it was invoked 0 times"
+      assert_raise Mox.VerificationError, message, &verify!/0
+
+      Task.Supervisor.async_nolink(MoxTests.TaskSupervisor, fn ->
+        CalcMock.add(2, 3)
+        CalcMock.add(4, 5)
+      end)
+      |> Task.yield()
```

Comment:
**Annotation:** Other places in this test file use a variable-based syntax instead of piping. For example: https://github.com/dashbitco/mox/blob/31217186dc7b618ea35448d2175377408a574fb3/test/mox_test.exs#L175:L181

I find this shorter, pipe-based syntax a bit nicer since it puts visual attention on the `async_nolink` call rather than on variable assignment, but I don't feel strongly about this choice. If you'd like me to change the syntax to match the existing code I'm happy to!